src/init.c

```diff
@@ -744,6 +744,8 @@ JL_DLLEXPORT void julia_init(JL_IMAGE_SEARCH rel)
 
     // initialize symbol-table lock
     uv_mutex_init(&symtab_lock);
+    // initialize profiler lock
+    uv_mutex_init(&profile_lock);
```

Comment:
😮 why wasn't this happening before?

Reply:
idk if we want to rename even more things in this PR, but...

the names in this block of code now seem a bit confusing
```julia
// initialize profiler lock
uv_mutex_init(&profile_lock);
// initialize backtraces
jl_init_profile_lock();
```
where `jl_init_profile_lock` and friends are defined here: https://github.com/JuliaLang/julia/blob/1f935afc07edde9f8c2e1a0f05d4772e18a55e97/src/signal-handling.c#L86-L145

Would it make sense to distinguish these more clearly, maybe the existing one should have a name like `debuginfo_lock`?

 or at least comments making clear their  different roles

Reply:
Yes, profile_lock is not a great name.

Let's rename it to bt_data_lock to make it more explicit what data structure it's protecting (and also to avoid the name confusion).